streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -469,7 +469,7 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
      */
     private def startReceiver(receiver: Receiver[_], scheduledExecutors: Seq[String]): Unit = {
       val receiverId = receiver.streamId
-      if (!isTrackerStarted) {
+      if (isTrackerStopping || isTrackerStopped) {
```

Comment:
Can you make this a internal function? `def shouldStartReceiver: Boolean = !(isTrackerStopping || isTrackerStopped)`
and then use `if (!shouldStartReceiver) { ... }`